### DISCUSSION TITLE
Adopt Lovelace config handling for automations/script integration

### DISCUSSION THREAD

**Discussion by balloob** [home-assistant/architecture#895] [General]
The current script/automations work by reading/writing directly to a YAML file. This is confusing for users, and comments/other YAML tags are getting lost. This is not a good experience.

Lovelace offers users two options to dealing with configuration:
- You can use YAML, including all the secrets and includes that YAML has to offer.
- You can use the storage helper, and have a UI to edit the configuration, including a raw YAML editor.

My proposal is to adopt this approach for the script and automation component and update the UI to also offer a raw YAML editor.

**Comment by pvizeli** [comment]
I would love those mechanics on all part of Home Assistant and take one red line over the whole platform.
That makes it available for End User and the DIY community.

For me 💯 x 💯

**Comment by elupus** [comment]
While its nice to edit in UI, being unable to versioncontrol is really really annoying. I wish there was some middle ground.

**Comment by balloob** [comment]
@elupus automatins/scripts are using a hybrid approach right now and it's not ideal. We have explored other options for Lovelace, those didn't work out either. 

I agree that version control is nice. Nothing will have to stop us from adding it to the storage version, including integration it in the UI. (but out of scope for this issue)

**Comment by Swamp-Ig** [comment]
This is out of scope for this particular issue, however it would be good to implement these as a 'module' kind of thing, much like node red does with tabs. That way you can group related automatons, scripts, and maybe even group definitions or devices together in the same module. 

Not part of this issue clearly but something to keep in mind in the design process so that we don't make implementing this in the future a major breaking change.

**Comment by balloob** [comment]
You mean areas? 😝

**Comment by Swamp-Ig** [comment]
Not entirely. I have a tab on note red for climate control, one for timed lights, and some others. They're not really areas since they have no geographic association.

**Comment by frenck** [comment]
While this might still be considered in the long term, this is not something we want to pursue right now. Therefore, going to close this proposal for now. We can always reopen and reconsider this proposal in the future.

../Frenck
